frontend/packages/policy-editor/src/components/PolicyCardRules/PolicyRule/PolicyAccessPackages/PolicyAccessPackages.tsx

```diff
@@ -0,0 +1,164 @@
+    return groupAccessPackagesByArea(accessPackages);
+  }, [accessPackages]);
+
+  const onPackageSelectChange = (accessPackage: PolicyAccessPackage): void => {
+    const isSelected = chosenAccessPackages.includes(accessPackage.urn);
+    if (isSelected) {
+      handleRemoveAccessPackage(accessPackage.urn);
+    } else {
+      handleAddAccessPackage(accessPackage.urn);
+    }
+  };
+
+  const handleRemoveAccessPackage = (packageUrn: string): void => {
+    setChosenAccessPackages((oldUrns) => oldUrns.filter((urn) => urn !== packageUrn));
+    const urnsToSave = policyRule.accessPackages.filter((x) => x !== packageUrn);
+
+    handleAccessPackageChange(urnsToSave);
+  };
+
+  const handleAddAccessPackage = (packageUrn: string): void => {
+    setChosenAccessPackages((oldUrns) => [...oldUrns, packageUrn]);
+    const urnsToSave = [...policyRule.accessPackages, packageUrn];
+
+    handleAccessPackageChange(urnsToSave);
+  };
+
+  const handleAccessPackageChange = (newSelectedAccessPackageUrns: string[]): void => {
+    const updatedRules = getUpdatedRules(
+      {
+        ...policyRule,
+        accessPackages: newSelectedAccessPackageUrns,
+      },
+      policyRule.ruleId,
+      policyRules,
+    );
+
+    setPolicyRules(updatedRules);
+    savePolicy(updatedRules);
+  };
+
+  const handleSearch = (search: string) => {
+    setSearchValue(search);
+  };
+
+  const isStringMatch = (matchString: string) => {
+    return matchString.toLowerCase().includes(searchValue.toLowerCase());
+  };
+
+  const accessPackagesToRender = groupedAccessPackagesByArea.reduce(
+    (areas: PolicyAccessPackageArea[], area): PolicyAccessPackageArea[] => {
+      const matchingPackages = area.packages.filter(
+        (pack) => !searchValue || isStringMatch(pack.name) || isStringMatch(pack.description),
+      );
+      const returnAreas = [...areas];
+      if (matchingPackages.length > 0) {
+        returnAreas.push({ ...area, packages: matchingPackages });
+      }
+      return returnAreas;
+    },
+    [],
+  );
+
+  const renderAccessPackageAccordion = (accessPackage: PolicyAccessPackage): React.ReactNode => {
+    const isChecked = chosenAccessPackages.includes(accessPackage.urn);
+    const checkboxLabel = t(
+      isChecked ? 'policy_editor.access_package_remove' : 'policy_editor.access_package_add',
+      {
+        packageName: accessPackage.name,
+      },
+    );
+    const packageCheckbox = (
+      <CheckboxGroup
+        legend=''
+        className={classes.accordionCheckbox}
+        value={isChecked ? [CHECKED_VALUE] : []}
+        onChange={() => onPackageSelectChange(accessPackage)}
+      >
+        <Checkbox value={CHECKED_VALUE} aria-label={checkboxLabel} />
+      </CheckboxGroup>
+    );
+    return (
+      <PolicyAccessPackageAccordion
+        key={accessPackage.urn}
+        accessPackage={accessPackage}
+        selectedLanguage={selectedLanguage}
+        selectPackageElement={packageCheckbox}
+      />
+    );
+  };
```

Comment:
I suggest splitting this up into smaller components outside this component to separate what each component does, and to increase readability